contracts/btc-staking/src/contract.rs

```diff
@@ -159,6 +160,28 @@ pub fn execute(
     }
 }
 
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn sudo(
```

Comment:
I'm wondering whether it's possible to let finality contract calls btc staking contract upon each `BeginBlock`, rather than let Babylon SDK call btc staking contract. This avoids this sudo call, which is imo pretty risky and needs to be minimised

Reply:
Hmm good point, yes the cross-contract call happens only at the end of the message execution. Then I don't think we have the other choice then :+1: 

Wdyt we have some docstring for documenting this?